### DISCUSSION TITLE
Generating colors from data, by RGB channels

### DISCUSSION THREAD

**Discussion by dcbb** [vega/altair#2956] [Q&A] [answered]
Is it possible to get a color for a mark by specifying three different numerical columns as red, green and blue components of the mark's color?

Something like:

```
alt.Chart(data).mark_point().encode(
   x="x", 
   y="y",
   color=alt.Color(red="r", green="g", blue="b")
)
```

...where `"r"`, `"g"`, and `"b"` are numerical columns in `data`.

Excessive googling suggests that it does not seem possible, but I'm hoping that I have overlooked something.

_Fun fact:_ I asked this to ChatGPT, and it very confidently suggests that there is an aggregate function in altair named `rgb`, so you can just go `Color("rgb(r,g,b)")`. Nice illustration of current AI limits.

**Comment by binste** [comment, marked as answer]
It is possible using so called expressions, see [Expressions for interaction](https://altair-viz.github.io/user_guide/.ipynb_checkpoints/interactions-checkpoint.html#expressions-for-interaction) for an introduction. ChatGPT was onto something but it's not quite correct :)

```python
import altair as alt
import pandas as pd

df = pd.DataFrame(
    [[1, 1, 102, 204, 0], [2, 2, 0, 0, 255], [3, 3, 0, 0, 0]],
    columns=["x", "y", "r", "g", "b"],
)

alt.Chart(df).mark_circle(size=200).encode(
    x="x",
    y="y",
    # In Altair 5 you will be able to replace alt.ExprRef with alt.expr
    color=alt.value(alt.ExprRef(alt.expr.rgb(alt.datum.r, alt.datum.g, alt.datum.b))),
    # Or you can write the expression as a string like this
    # color=alt.value(alt.ExprRef("rgb(datum.r, datum.g, datum.b)")),
)
```

<img width="453" alt="image" src="https://user-images.githubusercontent.com/23366411/224472565-d976b862-54bf-4c6b-8427-6c0e66d8e707.png">

**Reply by dcbb** [reply]
Works like a charm, thanks a lot! Now I don't have to revert to matplotlib for that plot.

Those "code in string" hacks are creepy but so powerful.
